### DISCUSSION TITLE
Allowing for more than 4 seconds in worker_process_init

### DISCUSSION THREAD

**Discussion by NixBiks** [celery/celery#7248] [Q&A]
According to the docs we are only allowed to have blocking code for a max of 4 seconds in [`worker_process_init`](https://docs.celeryproject.org/en/latest/userguide/signals.html#worker-process-init). I need some initial data before the worker should pick up any tasks and that takes more than 4 seconds. What do I do?

**Comment by nicolaspanel** [comment]
Hi @NixBiks 
have you tried adjust [PROC_ALIVE_TIMEOUT](https://github.com/celery/celery/blob/7b585138af8318d62b8fe7086df7e85d110ac786/celery/app/defaults.py#L324) or [CELERYD_WORKER_LOST_WAIT](https://github.com/celery/celery/blob/7b585138af8318d62b8fe7086df7e85d110ac786/celery/app/defaults.py#L318) (aka LOST_WAIT) parameters ?
